crates/dojo-lang/src/compiler.rs

```diff
@@ -525,7 +568,8 @@ fn get_dojo_model_artifacts(
         {
             // Leverages the contract selector function to only snake case the model name and
             // not the full path.
-            let contract_selector = ContractSelector(struct_id.full_path(db));
+            let contract_selector =
+                DojoContractSelector::new(struct_id.full_path(db), model.namespace.clone());
             let qualified_path = contract_selector.path_with_model_snake_case();
             let compiled_class = compiled_classes.get(&qualified_path).cloned();
             let tag = naming::get_tag(&model.namespace, &model.name);
```

Comment:
We should add some tracing log here to easily debug by having all the names/paths being printed in debug/trace mode, wdyt?

Reply:
Yes, good idea !